cli/src/semgrep/commands/ci.py

```diff
@@ -501,7 +503,15 @@ def ci(
     # place we wouldn't need this seprarately. There are some benefits
     # (e.g., separate progress bar), but it would simplify output logic if we
     # simply had one "scan".
-    if run_secrets and scan_handler and scan_handler.historical_config.enabled:
+    if not run_secrets and historical_secrets:
+        logger.info("Cannot run historical secrets scan without secrets enabled.")
+        sys.exit(FATAL_EXIT_CODE)
```

Comment:
Can you move this to the top of the function actually? Right now it is after the non historical scan.

Reply:
done in [3612e24](https://github.com/semgrep/semgrep/pull/9931/commits/3612e24218dd3d0b9cc09903bb18af702b4a82c3)